File: 정민선/프로그래머스/[1차] 캐시.py
def solution(cacheSize, cities):
    answer = 0
    cities = [i.upper() for i in cities]
    if cacheSize == 0:
        return len(cities)*5
    
    cache = []
    for city in cities:
        if city in cache:
            answer += 1
            cache.remove(city)
            cache.append(city)
        else:
            answer += 5
            if cacheSize > len(cache):
                cache.append(city)
            else:
                cache.pop(0)
                cache.append(city)

    return answer

File: 이진영/프로그래머스/[1차] 캐시.js
function solution(cacheSize, cities) {
    const CACHE_HIT = 1;
    const CACHE_MISS = 5;
    
    if (cacheSize === 0) return cities.length * CACHE_MISS;
    
    let page = Array(cacheSize).fill(null);
    let t = 0;
    
    for (let city of cities) {
        const lCity = city.toLowerCase();
        if (!page.includes(lCity)) {
            t += CACHE_MISS;
            page.shift();
            page.push(lCity);
        } else {
            const idx = page.indexOf(lCity);
            page = [...page.slice(0, idx), ...page.slice(idx + 1), lCity];
            t += CACHE_HIT;   
        }
    }
    return t;
}